### PR TITLE
#14705 Fix edit not working properly in EditMavenArtifactDialog

### PR DESCRIPTION
**Pull request by Destrolaric**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ protected Composite createDialogArea(Composite parent) {
                 public void widgetSelected(SelectionEvent e) {
                     artifacts.clear();
                     //check if we are on raw tab. When trues to parse
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java]: `When trues to parse` what.

This comment is useless anyway. The code on its own perfectly describes what's going on.

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java [hunk 2]
```diff
@@ -361,6 +363,11 @@ public void saxText(SAXReader reader, String data) {
 
     }
 
+    private enum TabData{
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java]:
```suggestion
    private enum TabType {
```

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java [hunk 3]
```diff
@@ -361,6 +363,11 @@ public void saxText(SAXReader reader, String data) {
...
+        RAW,
+        MANUAL
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java]: We don't use these terms in this dialogue anymore. After some time, we will forget them and will be pretty confused when reading the code that uses these constants. Please consider renaming.

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java [hunk 4]
```diff
@@ -105,16 +105,15 @@ protected Composite createDialogArea(Composite parent) {
                 @Override
                 public void widgetSelected(SelectionEvent e) {
                     artifacts.clear();
-                    //check if we are on raw tab. When trues to parse
-                    if (tabFolder.getSelectionIndex() == 0 && tabFolder.getItemCount() != 1){
+                    if (tabFolder.getSelection()[0].getData() == TabType.DEPENDENCY_DECLARATION){
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java:108]:
```suggestion
                    if (tabFolder.getSelection()[0].getData() == TabType.DEPENDENCY_DECLARATION) {
```

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java [hunk 5]
```diff
@@ -225,6 +225,7 @@ private void createManualTab(@NotNull TabFolder folder) {
         TabItem item = new TabItem(folder, SWT.NONE);
         item.setText(UIConnectionMessages.dialog_edit_driver_edit_maven_manual);
         item.setControl(container);
+        item.setData(TabType.DECLARE_ARTIFACT_MANUALLY);
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java:230]:
```suggestion

```

**Pull request merged**